digdag-spi/src/main/java/io/digdag/spi/Storage.java

```diff
@@ -35,4 +35,8 @@ String put(String key, long contentLength,
     {
         return Optional.absent();
     }
+
+    Long getDirectDownloadExpiration();
+
+    Long getDirectUploadExpiration();
```

Comment:
To keep compatibility, it is better to change to default method.
In addition, I guess it is better that the return value is Optional<Long> like getDirectDownloadHandle(), getDirectUploadHandle().

Reply:
In addition add comments on unit of return value is good. (seconds? or milliseconds?)

Reply:
thanks for your review.
Updat to use default method and use Optinal type.